src/Core/Case/Util.idr

```diff
@@ -0,0 +1,62 @@
+module Core.Case.Util
+
+import Core.Case.CaseTree
+import Core.Context
+import Core.Env
+import Core.Normalise
+import Core.Value
+
+||| Given a normalised type, get all the possible constructors for that
+||| type family, with their type, name, tag, and arity.
+export
+getCons : {auto c : Ref Ctxt Defs} ->
+          {vars : _} ->
+          Defs -> NF vars -> Core (List (NF [], Name, Int, Nat))
```

Comment:
Can we introduce a `Constructor` record that gives suggestive field names for
these `(NF [], Name, Int, Nat)` values? I'm worried of the over-abundance of
cryptic tuples in the compiler.

Reply:
I like this idea a lot. I've also felt encumbered by needing to decipher tuples in parts of the compiler code before.

Reply:
Done. And in the process I discovered that only 3 of the 4 elements of the tuple were needed.